timefold-solver/src/test/java/com/baeldung/timefoldsolver/ShiftScheduleConstraintProviderTest.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.timefoldsolver;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+import ai.timefold.solver.test.api.score.stream.ConstraintVerifier;
+import org.junit.jupiter.api.Test;
+
+class ShiftScheduleConstraintProviderTest {
+
+    private static final LocalDate MONDAY = LocalDate.of(2030, 4, 1);
+    private static final LocalDate TUESDAY = LocalDate.of(2030, 4, 2);
+
+    ConstraintVerifier<ShiftScheduleConstraintProvider, ShiftSchedule> constraintVerifier = ConstraintVerifier.build(
+            new ShiftScheduleConstraintProvider(), ShiftSchedule.class, Shift.class);
+
+    @Test
+    void atMostOneShiftPerDay() {
+        Employee ann = new Employee("Ann", null);
+        constraintVerifier.verifyThat(ShiftScheduleConstraintProvider::atMostOneShiftPerDay)
+                .given(
+                        ann,
+                        new Shift(MONDAY.atTime(6, 0), MONDAY.atTime(14, 0), null, ann),
+                        new Shift(MONDAY.atTime(14, 0), MONDAY.atTime(22, 0), null, ann))
+                // Penalizes both A-B and B-A. To avoid that, use forEachUniquePair() in the constraint instead.
```

Comment:
As someone new to this topic, I find this a bit cryptic. Is there a simpler way to explain what you were communicating here?

Reply:
Yea, I struggle with explaining this one :(
Basically, the score penalty is 2 instead of 1 because it penalizes both the tuple {shift A, shift B} and the tuple {shift B, shift A}. The fix entails adding PlanningId annotations which I didn't want to bring into scope.

Let me try rewording that.

Reply:
What do you think of this?

```
                // Penalizes by 2 because both {shiftA, shiftB} and {shiftB, shiftA} match.
                // To avoid that, use forEachUniquePair() in the constraint instead.
```

Reply:
>  // Penalizes by 2 because both {shiftA, shiftB} and {shiftB, shiftA} match.
Makes sense

> // To avoid that, use forEachUniquePair() in the constraint instead.
Meaning, forEachUniquePair will consider {shiftA, shiftB} as equivalent to {shiftB, shiftA}?

It would help me if we could say "To avoid that, use forEachUniquePair() in the constraint instead of ___ in the constraint instead", it wasn't clear which part of the constraint forEachUniquePair would replace

Reply:
@ge0ffrey this one I'd like clarified please